### DISCUSSION TITLE
FREQ: Add Entry Type to View field selection list (particularly for Custom Sources) or combine View for Fields of multiple Entry Types to one Column

### DISCUSSION THREAD

**Discussion by humandevs** [craftcms/cms#16119] [Ideas] [answered]
The list of fields to add to a Custom Sources view can be confusing as it lists all fields even if a View filter limits the Entry Types to include in that view, and because the same Field can now be used in multiple Entry Types, all of them are listed in the View columns, and there's no easy way to differentiate which Entry Type corresponds to the Field added to the View. Adding the Entry Type to the label for the Field in the View list would solve this issue. 

Example screenshot of multiple Fields with the same name but for different Entry Types confusingly presented as options for adding to a Custom Sources view:

![image](https://github.com/user-attachments/assets/313a6861-058e-46b0-95f7-f7d69abbb011)

Even better would be to only need to add the Field column once for all Entry Types Fields that are of the same source Field type, so that multiple columns don't have to be taken up to show contents of Fields of the same source/name but from multiple Entry Types, but that's a whole other can of worms!

**Comment by brandonkelly** [comment]
Are those actually all the same custom field? Or do you have multiple fields with the same name?

**Reply by humandevs** [reply]
Ahah! This looks like a bug. These are all one custom Field type, added to multiple Entry Types.

- Before saving newly created Custom Source = duplicates
- When editing an existing Custom Source = no duplicates

When you create a new Custom Source, *before* you save it, you get these duplicate fields listed for each Entry Type that has one. As soon as the Custom Source is saved, if you edit the Custom Source, only one column per custom Field is shown, regardless of the number of Entry Types that custom Field is a part of. Happens if no filter is applied and/or an Entry Type filter is applied for me on Craft 5.5.0.1

**Reply by humandevs** [reply]
I have created issue https://github.com/craftcms/cms/issues/16177 to track this

**Comment by brandonkelly** [comment, marked as answer]
Closing in favor of #16177.
